### PR TITLE
adding extensionAttributes for devices

### PR DESCRIPTION
**Pull request by SanDeo-MSFT**
Updated devices and added a new extensionAttributes.md file

### CODE REVIEW

File: api-reference/beta/resources/extensionattributes.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+title: "extensionAttributes resource type"
+description: "The **extensionAttributes** property of the device entity contains fifteen custom extension attribute properties. This set of properties is mastered in cloud and may be set during creation or update of a device."
```
Comment (dkershaw10) [inline api-reference/beta/resources/extensionattributes.md:3]: Remove second sentence in the description header.

File: api-reference/beta/resources/extensionattributes.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+author: "sandeo"
+---
+
+# extensionAttributes resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+The **extensionAttributes** property of the [device](device.md) entity contains fifteen custom extension attribute properties. This set of properties is mastered in cloud and may be set during creation or update of a device.
```
Comment (dkershaw10) [inline api-reference/beta/resources/extensionattributes.md:16]: entity -> resource

File: api-reference/beta/resources/extensionattributes.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      "extensionAttribute11": "string",
+      "extensionAttribute12": "string",
+      "extensionAttribute13": "string",
+      "extensionAttribute14": "string",
+      "extensionAttribute15": "string"
+  }
+}
+```
+
+### Use case: Bad request when using filter on extensionAttributes
```
Comment (dkershaw10) [inline api-reference/beta/resources/extensionattributes.md:84]: Please remove - I don't think we need a use case for this.  You already explained elsewhere that this is not supported and 400 BadRequest is already in our error messages topic.

File: api-reference/v1.0/resources/device.md [hunk 4]
```diff
@@ -45,6 +45,7 @@ This resource lets you add your own data to custom properties using [extensions]
 |deviceMetadata|String| For interal use only. Set to null. |
 |deviceVersion|Int32| For interal use only. |
 |displayName|String|The display name for the device. Required. |
+| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. Note that the individual extension attributes are neither selectable nor filterable. This set of properties is mastered in cloud and may be set during creation or update of a device. <br><br>Returned only on $select. |
```
Comment (dkershaw10) [inline api-reference/v1.0/resources/device.md:48]: You could also mention here that this is not filterable.

File: api-reference/beta/resources/device.md [hunk 5]
```diff
@@ -47,6 +47,7 @@ This resource lets you add your own data to custom properties using [extensions]
 |deviceMetadata|String| For internal use only. Set to null. |
 |deviceVersion|Int32| For internal use only. |
 |displayName|String| The display name for the device. Required. |
+| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. Note that the individual extension attributes are neither selectable nor filterable. This set of properties is mastered in cloud and may be set during creation or update of a device. <br><br>Returned only on $select. |
```
Comment (davidmu1) [inline api-reference/beta/resources/device.md:50]: Contains extension attributes 1-15 for the device. The individual extension attributes are not selectable or filterable. This set of properties is mastered in the cloud and may be set during creation or update of a device. Returned only on $select.

File: api-reference/beta/resources/extensionattributes.md (hunk 1 above)
Comment (SanDeo-MSFT) [reply api-reference/beta/resources/extensionattributes.md:3]: Dan, The sentence is there to indicate that this property can only be set in AAD and not like users where you can set it on prem and sync to AAD.

File: api-reference/beta/resources/extensionattributes.md (hunk 2 above)
Comment (SanDeo-MSFT) [reply api-reference/beta/resources/extensionattributes.md:16]: I used the word entity to match the docs we have for user.

File: api-reference/v1.0/resources/device.md (hunk 4 above)
Comment (SanDeo-MSFT) [reply api-reference/v1.0/resources/device.md:48]: It is already mentioned.

File: api-reference/beta/resources/device.md (hunk 5 above)
Comment (SanDeo-MSFT) [reply api-reference/beta/resources/device.md:50]: Will do

File: api-reference/beta/resources/extensionattributes.md (hunk 1 above)
Comment (dkershaw10) [reply api-reference/beta/resources/extensionattributes.md:3]: The header is used for search.  Your full description is already in the main MD doc.  Please remove the second sentence from the header.

File: api-reference/beta/resources/extensionattributes.md (hunk 2 above)
Comment (dkershaw10) [reply api-reference/beta/resources/extensionattributes.md:16]: Well that needs fixing too.  Please change to "resource"

File: api-reference/beta/resources/device.md [hunk 6]
```diff
@@ -47,6 +47,7 @@ This resource lets you add your own data to custom properties using [extensions]
 |deviceMetadata|String| For internal use only. Set to null. |
 |deviceVersion|Int32| For internal use only. |
 |displayName|String| The display name for the device. Required. |
+| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. The individual extension attributes are neither selectable nor filterable. This set of properties is mastered in cloud and may be set during creation or update of a device. <br><br>Returned only on $select. |
```
Comment (dkershaw10) [inline api-reference/beta/resources/device.md:50]: Not sure the filterable is correct now.  I believe that Mezzo now allows filtering on extensionAttribute individual items - at least it works on user.

**Comment by SanDeo-MSFT** [comment]
ready to merge

File: api-reference/beta/resources/device.md (hunk 6 above)
Comment (davidmu1) [inline api-reference/beta/resources/device.md:50]:
```suggestion
| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. The individual extension attributes are not selectable or filterable. This set of properties may be set during creation or update of a device. <br><br>Returned only on $select. |
```

File: api-reference/beta/resources/extensionattributes.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
+---
+title: "extensionAttributes resource type"
+description: "The **extensionAttributes** property of the device entity contains fifteen custom extension attribute properties."
```
Comment (davidmu1) [inline api-reference/beta/resources/extensionattributes.md:3]:
```suggestion
description: "The extensionAttributes property of the device entity contains fifteen custom extension attribute properties."
```

File: api-reference/beta/resources/extensionattributes.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+author: "sandeo"
+---
+
+# extensionAttributes resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+The **extensionAttributes** property of the [device](device.md) resource contains fifteen custom extension attribute properties. This set of properties is mastered in cloud and may be set during creation or update of a device.
```
Comment (davidmu1) [inline api-reference/beta/resources/extensionattributes.md:16]:
```suggestion
The **extensionAttributes** property of the [device](device.md) resource contains fifteen custom extension attribute properties. This set of properties may be set during creation or update of a device.
```

File: api-reference/beta/resources/extensionattributes.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+|extensionAttribute8|String| Eighth customizable extension attribute. |
+|extensionAttribute9|String| Ninth customizable extension attribute. |
+|extensionAttribute10|String| Tenth customizable extension attribute. |
+|extensionAttribute11|String| Eleventh customizable extension attribute. |
+|extensionAttribute12|String| Twelfth customizable extension attribute. |
+|extensionAttribute13|String| Thirteenth customizable extension attribute. |
+|extensionAttribute14|String| Fourteenth customizable extension attribute. |
+|extensionAttribute15|String| Fifteenth customizable extension attribute. |
+
+## Example REST operations
```
Comment (davidmu1) [inline api-reference/beta/resources/extensionattributes.md:38]: This section should not be here. If needed these examples should be added to the appropriate method pages, such as Update Device or Get Device.

File: api-reference/v1.0/resources/device.md [hunk 10]
```diff
@@ -45,6 +45,7 @@ This resource lets you add your own data to custom properties using [extensions]
 |deviceMetadata|String| For interal use only. Set to null. |
 |deviceVersion|Int32| For interal use only. |
 |displayName|String|The display name for the device. Required. |
+| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. The individual extension attributes are neither selectable nor filterable. This set of properties is mastered in cloud and may be set during creation or update of a device. <br><br>Returned only on $select. |
```
Comment (davidmu1) [inline api-reference/v1.0/resources/device.md:48]:
```suggestion
| extensionAttributes | [extensionAttributes](extensionattributes.md) | Contains extensionAttributes 1-15 for the device. The individual extension attributes are neither selectable nor filterable. This set of properties may be set during creation or update of a device. <br><br>Returned only on $select. |
```

File: api-reference/v1.0/resources/extensionattributes.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
+---
+title: "extensionAttributes resource type"
+description: "The **extensionAttributes** property of the device entity contains fifteen custom extension attribute properties."
```
Comment (davidmu1) [inline api-reference/v1.0/resources/extensionattributes.md:3]:
```suggestion
description: "The extensionAttributes property of the device entity contains fifteen custom extension attribute properties."
```

File: api-reference/v1.0/resources/extensionattributes.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+author: "sandeo"
+---
+
+# extensionAttributes resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+The **extensionAttributes** property of the [device](device.md) resource contains fifteen custom extension attribute properties. This set of properties is mastered in cloud and may be set during creation or update of a device.
```
Comment (davidmu1) [inline api-reference/v1.0/resources/extensionattributes.md:16]:
```suggestion
The **extensionAttributes** property of the [device](device.md) resource contains fifteen custom extension attribute properties. This set of properties may be set during creation or update of a device.
```

File: api-reference/v1.0/resources/extensionattributes.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+|extensionAttribute8|String| Eighth customizable extension attribute. |
+|extensionAttribute9|String| Ninth customizable extension attribute. |
+|extensionAttribute10|String| Tenth customizable extension attribute. |
+|extensionAttribute11|String| Eleventh customizable extension attribute. |
+|extensionAttribute12|String| Twelfth customizable extension attribute. |
+|extensionAttribute13|String| Thirteenth customizable extension attribute. |
+|extensionAttribute14|String| Fourteenth customizable extension attribute. |
+|extensionAttribute15|String| Fifteenth customizable extension attribute. |
+
+## Example REST operations
```
Comment (davidmu1) [inline api-reference/v1.0/resources/extensionattributes.md:38]: This section shouldn't be here. These examples should be added to the method pages.

File: api-reference/v1.0/resources/extensionattributes.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+``` JSON
+GET https://graph.microsoft.com/v1.0/devices?$filter=extensionAttributes/extensionAttributes1 eq 'extensionAttributes1-value'
+
+HTTP/1.1 400 BadRequest
+{
+    "Error message body"
+}
+```
+
```
Comment (davidmu1) [inline api-reference/v1.0/resources/extensionattributes.md:94]: Missing the JSON representation.

File: api-reference/beta/resources/extensionattributes.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+``` JSON
+GET https://graph.microsoft.com/beta/devices?$filter=extensionAttributes/extensionAttributes1 eq 'extensionAttributes1-value'
+
+HTTP/1.1 400 BadRequest
+{
+    "Error message body"
+}
+```
+
```
Comment (davidmu1) [inline api-reference/beta/resources/extensionattributes.md:94]: Missing the JSON representation.

**Comment by davidmu1** [comment]
I think there also needs to be changelog entries added.

File: api-reference/beta/resources/extensionattributes.md (hunk 1 above)
Comment (SanDeo-MSFT) [reply api-reference/beta/resources/extensionattributes.md]: done

**Pull request closed without merging**